### PR TITLE
* 404 error page.

### PR DESCRIPTION
**Pull request by NurElHuda**
Add a component that handles 404 error.
Implements #107 

This is a screenshot of the result:
image

I was going for a more clean design like: 
image

However, It needs refactoring of the Main component with its Switch and Routing. So, do I go all the way? or do I leave like this?

Another issue is that it only catches unfound paths that did not get matched by previous routs. So, a path like "/Learn/Git_Bas" or "/Articles/Send_Emails_With_D" does not get handled. Any ideas on how should I handle that?

## Checklist

- [ ] Affected `data`
- [x] Affected `frontend`

### CODE REVIEW

**Review by ZibanPirate — CHANGES_REQUESTED** [review]
The current implementation is fine, still look clean at the same time user still can navigate to any section of the website.

as for the nested paths issue, this is out of this PR scope, since when we have wrong article link, we should redirect to `/Articles` url, and add a landing page for it, that being said, we can still fix it as part of #107 

the experience will be simular to zakiii.com
where going to
https://www.zakiii.com/Articles/My_2019_in_Review
is fine, but when going to a wrong link
https://www.zakiii.com/Articles/My_2020_in_Review
will redirect you to 
https://www.zakiii.com/Articles
with a nice landing page that tell you to browse articles from the sidebar.

for doing this, we need to change two things
- when we fetch the article, when call fails or we get a 404 response, we force redirect to `/Articles`, that can be done [here](https://github.com/dzcode-io/dzcode.io/blob/master/frontend/src/apps/main/redux/actions/articles-page/index.ts) where we check if response if good, if not we throw an exception [here](https://github.com/dzcode-io/dzcode.io/blob/master/frontend/src/apps/main/redux/actions/articles-page/index.ts#L74), and lastly we redirect to `/Articles` on the `catch` block [here](https://github.com/dzcode-io/dzcode.io/blob/master/frontend/src/apps/main/redux/actions/articles-page/index.ts#L86)
- second thing to do is adding a landing page for `/Articles`, that can be done [here](https://github.com/dzcode-io/dzcode.io/blob/master/frontend/src/apps/main/pages/articles/index.tsx#L41) where we add a sub-page similar to [content](https://github.com/dzcode-io/dzcode.io/tree/master/frontend/src/apps/main/pages/articles/content) sub-page but with a static content instead

Both these points can be done separately in two PRs as part of #107 , i can help out in one of them or you can do both, the exact same thing applies for `/Learn` btw.

File: frontend/src/apps/main/entry/index.tsx [hunk 1]
```diff
@@ -27,6 +27,7 @@ const Projects = lazy(() => import("t9/apps/main/pages/projects"));
 const Learn = lazy(() => import("t9/apps/main/pages/learn"));
 const Faq = lazy(() => import("t9/apps/main/pages/faq"));
 const Contact = lazy(() => import("t9/apps/main/pages/contact"));
+const NotFound = lazy(() => import("t9/apps/main/pages/notFound"));
```
[the remark this answers is not in the document]
Comment (NurElHuda) [reply frontend/src/apps/main/entry/index.tsx]: I've renamed the file to be snaked case.

**Comment by NurElHuda** [comment]
I agree with separating the PRs. 
I will take care of them both.

**Review by ZibanPirate — APPROVED** [review]
That also works 👌🏽 thanks for the changes @NurElHuda

**Pull request merged**